src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/UnsupportedTypeConverterFactory.cs

```diff
@@ -1,13 +1,19 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
 using System.Reflection;
 using System.Runtime.Serialization;
 
 namespace System.Text.Json.Serialization.Converters
 {
     internal sealed class UnsupportedTypeConverterFactory : JsonConverterFactory
     {
+        [RequiresDynamicCode("JSON serialization and deserialization might require types that cannot be statically analyzed and might need runtime code generation. Use System.Text.Json source generation for native AOT applications.")]
```

Comment:
Can this use SerializationRequiresDynamicCodeMessage constant?

(multiple places)